### ISSUE TITLE
Tags pagination is not working

### ISSUE THREAD

**Issue by nboMikeLammers** [mailgun/mailgun-php#700]
I am trying to get other pages from my tags response but I get the error `Uncaught Error: Call to undefined method Mailgun\Api\Tag::nextPage()` Any clue?

```php
$mgClient = Mailgun::create('key----------------------');

$response = $mgClient->tags()->index($this->mgDomain,10);
$response2 = $mgClient->tags()->nextPage($response);
var_dump($response2);
var_dump($response);
```

**Comment by DavidGarciaCat** [comment]
Yes, here's the reason:

- [src/Api/Tag.php](https://github.com/mailgun/mailgun-php/blob/master/src/Api/Tag.php) provides the functionality that handles `Tags`, and this class extends from [src/Api/HttpApi.php](https://github.com/mailgun/mailgun-php/blob/master/src/Api/HttpApi.php).

- Neither `Tags` nor `HttpApi` include by default the [src/Api/Pagination.php](https://github.com/mailgun/mailgun-php/blob/master/src/Api/Pagination.php) `Trait`, so the methods to browse other pages of the result are not available due to this reason.

You are more than welcome to send a Pull Request, including the required fix and PHP Unit tests. 😄 

Cheers,

**Comment by krismanning** [comment]
I'm a bit confused on this one. The documentation itself uses tags as the example for pagination....

`/** @var Mailgun\Model\Tag\IndexReponse $response */
$response = $mailgun->tags()->index('example.com');

// Parse through the first response
// ...

$nextResponse = $mailgun->tags()->nextPage($response);
$previousResponse = $mailgun->tags()->previousPage($response);
$firstResponse = $mailgun->tags()->firstPage($response);
$lastResponse = $mailgun->tags()->lastPage($response);`

**Comment by DavidGarciaCat** [comment]
Can you please provide the URL to check and compare?

It's my guess that the documentation must not be up to date, but also the fact that the SDK has this missing implementation given the examples that we can see on the API Docs page

https://documentation.mailgun.com/en/latest/api-tags.html#examples

![image](https://user-images.githubusercontent.com/7809429/95362096-82256380-08c5-11eb-91b4-e6acbf1343dc.png)

**Comment by krismanning** [comment]
The problem I've seen is whilst the pagination links are exposed the nextPage() etc methods don't work. I've had a play and it appears to be as simple as adding use Pagination; to the relevant API class.  For instance....

```
class MailingList extends HttpApi
{

    use Pagination;
```

I've never submitted a PR before but would be happy to give it a try if you wanted.

**Comment by DavidGarciaCat** [comment]
Thet's correct, there's just one missing addition (but don't remember if it is in the `Model` or the `API`).

You are more than welcome if you want to send a PR, just please remember adding PHP Unit tests too, to confirm the new code is working and we don't break previous functionalities.

**Comment by krismanning** [comment]
Thanks David,

It works when added to the relevant API class. I don't think it worked when added to the model though.

Let me take a look at PR's, it'll be useful learning for me.

Stay safe,

K

**Comment by Nyholm** [comment]
This was fixed by #717
